arch/riscv/core/irq_manage.c

```diff
@@ -19,15 +19,15 @@ LOG_MODULE_DECLARE(os, CONFIG_KERNEL_LOG_LEVEL);
 
 FUNC_NORETURN void z_irq_spurious(const void *unused)
 {
-	unsigned long mcause;
+	unsigned long xcause;
 
 	ARG_UNUSED(unused);
 
-	mcause = csr_read(mcause);
+	xcause = csr_read(xcause);
 
-	mcause &= CONFIG_RISCV_MCAUSE_EXCEPTION_MASK;
+	xcause &= CONFIG_RISCV_MCAUSE_EXCEPTION_MASK;
 
-	LOG_ERR("Spurious interrupt detected! IRQ: %ld", mcause);
+	LOG_ERR("Spurious interrupt detected! IRQ: %ld", xcause);
 #if defined(CONFIG_RISCV_HAS_PLIC)
 	if (mcause == RISCV_IRQ_MEXT) {
```

Comment:
should be `if (xcause == RISCV_IRQ_MEXT)`?

Reply:
I m little cautious about things, I cant test myself. theoretically that you are correct. but i dont have a device to test that scenario.

should I still change it?

btw there might be more sources with options that I may not be able to test myself.

Reply:
> I m little cautious about things, I cant test myself. theoretically you are correct. but i dont have a device to test that scenario.
> Aye, I was cautious while pointing it out :). I'm not sure if the logic makes perfect sense, just yet. FWIW I'm testing this on a PolarFire SoC Icicle Kit. It complained while compiling so I pointed the above out.

The icicle kit has a monitor processor thats running in supervisor mode. I am testing your changes out on one of the 4 application cores. I am hitting a bunch of issues with interrupts very early in the startup... when I can see why they are happening a little more clearly I will report back